File: 0189 rotate-array/solution.cs
public class Solution {
    public void Rotate(int[] nums, int k) {
        k=k%nums.Length;
        rev(nums,0,nums.Length-1);
        if(k>0)
        rev(nums,0,k-1);
        rev(nums,k,nums.Length-1);
    }
    private void rev(int[] nums,int i,int j) {
        while(i<j) {
            int temp=nums[i];
            nums[i++]=nums[j];
            nums[j--]=temp;
        }
    }
}

File: 0189 rotate-array/solution.rs
impl Solution {
    pub fn rotate(nums: &mut Vec<i32>, k: i32) {
     let n = nums.len();
     let k=(k as usize)%n;
     Self::rev(nums,0,n-1);
     if k>0 {
        Self::rev(nums,0,k-1);
     }
     
     Self::rev(nums,k,n-1);
    }

    fn rev(nums: &mut Vec<i32>,mut i: usize, mut j: usize) {
        while i<j {
            nums.swap(i,j);
            i+=1;
            j-=1;
        }
    }
}
